File: sdkwork-ai-sdk-python/sdkwork_ai_sdk/api/image.py
from typing import Any, Dict, List, Optional
from ..http_client import HttpClient
from ..models import CreateImageEditPostRequest, CreateImageVariationRequest, ImageGenerationRequest, ImageGenerationResponse

class ImageApi:
    """image API client."""
    
    def __init__(self, client: HttpClient):
        self._client = client

    def create_image_variation(self, body: Optional[CreateImageVariationRequest] = None, params: Optional[Dict[str, Any]] = None) -> ImageGenerationResponse:
        """Create image variation"""
        return self._client.post(f"/ai/v3/images/variations", data=body, params=params)

    def create_image(self, body: ImageGenerationRequest) -> ImageGenerationResponse:
        """Create image"""
        return self._client.post(f"/ai/v3/images", json=body)

    def create_image_generations(self, body: ImageGenerationRequest) -> ImageGenerationResponse:
        """Create image"""
        return self._client.post(f"/ai/v3/images/generations", json=body)

    def create_image_edit(self, body: Optional[CreateImageEditPostRequest] = None, params: Optional[Dict[str, Any]] = None) -> ImageGenerationResponse:
        """Create image edit"""
        return self._client.post(f"/ai/v3/images/edits", data=body, params=params)

File: sdkwork-ai-sdk-go/api/image.go
package api

import (
    sdktypes "github.com/sdkwork/ai-sdk/types"
    sdkhttp "github.com/sdkwork/ai-sdk/http"
)

type ImageApi struct {
    client *sdkhttp.Client
}

func NewImageApi(client *sdkhttp.Client) *ImageApi {
    return &ImageApi{client: client}
}

// Create image variation
func (a *ImageApi) CreateImageVariation(body *sdktypes.CreateImageVariationRequest, query map[string]interface{}) (sdktypes.ImageGenerationResponse, error) {
    raw, err := a.client.Post(AiApiPath("/images/variations"), body, query, nil, "multipart/form-data")
    if err != nil {
        var zero sdktypes.ImageGenerationResponse
        return zero, err
    }
    return decodeResult[sdktypes.ImageGenerationResponse](raw)
}

// Create image
func (a *ImageApi) CreateImage(body sdktypes.ImageGenerationRequest) (sdktypes.ImageGenerationResponse, error) {
    raw, err := a.client.Post(AiApiPath("/images"), body, nil, nil, "")
    if err != nil {
        var zero sdktypes.ImageGenerationResponse
        return zero, err
    }
    return decodeResult[sdktypes.ImageGenerationResponse](raw)
}

// Create image
func (a *ImageApi) CreateImageGenerations(body sdktypes.ImageGenerationRequest) (sdktypes.ImageGenerationResponse, error) {
    raw, err := a.client.Post(AiApiPath("/images/generations"), body, nil, nil, "")
    if err != nil {
        var zero sdktypes.ImageGenerationResponse
        return zero, err
    }
    return decodeResult[sdktypes.ImageGenerationResponse](raw)
}

// Create image edit
func (a *ImageApi) CreateImageEdit(body *sdktypes.CreateImageEditPostRequest, query map[string]interface{}) (sdktypes.ImageGenerationResponse, error) {
    raw, err := a.client.Post(AiApiPath("/images/edits"), body, query, nil, "multipart/form-data")
    if err != nil {
        var zero sdktypes.ImageGenerationResponse
        return zero, err
    }
    return decodeResult[sdktypes.ImageGenerationResponse](raw)
}
